R/FilteredDataset-utils.R

```diff
@@ -1,22 +1,30 @@
-#' Initializes `FilteredDataset`
+#' Initialize `FilteredDataset`
 #'
-#' @param dataset (`data.frame` or `MultiAssayExperiment`)
-#' @param dataname (`character`)
-#'  A given name for the dataset it may not contain spaces
-#' @param keys optional, (`character`)
-#'   Vector with primary keys
+#' Initializes a `FilteredDataset` object corresponding to the class of the filtered dataset.
+#'
+#' @param dataset any object
+#' @param dataname (`character(1)`)
+#'  syntactically valid name given to the dataset.
+#' @param keys (`character`)
+#'   optional vector of primary key column names.
 #' @param parent_name (`character(1)`)
-#'   Name of the parent dataset
+#'   name of the parent dataset.
 #' @param parent (`reactive`)
-#'   object returned by this reactive is a filtered `data.frame` from other `FilteredDataset`
-#'   named `parent_name`. Consequence of passing `parent` is a `reactive` link which causes
-#'   causing re-filtering of this `dataset` based on the changes in `parent`.
+#'   that returns a filtered `data.frame` from other `FilteredDataset` named `parent_name`.
+#'   Passing `parent` results in a `reactive` link that causes re-filtering of this `dataset`
+#'   based on the changes in `parent`.
 #' @param join_keys (`character`)
-#'   Name of the columns in this dataset to join with `parent`
-#'   dataset. If the column names are different if both datasets
-#'   then the names of the vector define the `parent` columns.
-#' @param label (`character`)
-#'   Label to describe the dataset
+#'   vector of names of columns in this dataset to join with `parent` dataset.
+#'   If column names in the parent do not match these, they should be given as the names of this vector.
+#' @param label (`character(1)`)
+#'   label to describe the dataset.
+#'
+#' @return Object of class `FilteredDataset`.
+#'
+#' @note
+#' Although this function is exported for use in other packages, it may be changed or removed in a future release,
+#' at which point any code which relies on this exported function will need to be changed.
```

Comment:
This is an universal truth - you probably don't have to add it here.

Reply:
ahh - spotted this
you actually moved this upper (not add it)
but still - this doesn't add much value IMHO

Reply:
I think this is supposed to mean that even though this function is exported, breaking changes may occur without warning, like in an internal one.

Reply:
I guess that after we will release on cran that statement is not necessarily true anymore :)

Reply:
The thing is the generic must be exported in order to extend the filter panel in `teal.modules.helios`.

We are trying to remove these exports in https://github.com/insightsengineering/teal.slice/issues/539 but I'm not sure we will be successful.
There is also the looming refactor of `teal.slice` in https://github.com/insightsengineering/teal.slice/pull/498.

Perhaps rename "Note" to "Warning" and add that it should only be used after consulting the maintainer?

Reply:
e9e2e9a90aa05b6748e385822768f2f4c4521a4f